src/js/services/link_track.js

```diff
@@ -0,0 +1,186 @@
+
+/**
+ * Compulsory fields for send event on clicking a link.
+ */
+function compulsoryFields(/* link */) {
+  return {
+    transport: 'beacon',
+  };
+}
+
+/**
+ * Returns the URL origin given a string representation of it. If the browser
+ * does not support URL, it returns href itself.
+ *
+ * @param {String} href the string containing the URL.
+ * @return {String} the origin of the URL.
+ */
+function getOrigin(href) {
+  if (URL !== undefined) {
+    return new URL(href).origin;
+  }
+
+  return href;
+}
+
+/**
+ * Computes the default values for some keys to the send event command.
+ *
+ * @param {HTMLAnchorElement} link anchor clicked.
+ * @return {Object} object with the default values.
+ */
+function defaultsForLink(link) {
+  const sameOrigin = getOrigin(link) === window.location.origin;
+  return {
+    eventCategory: sameOrigin ? 'inbound' : 'outbound',
+    eventAction: 'click',
+    eventLabel: link.href,
+  };
+}
+
+/**
+ * Computes the data for a given anchor element
+ *
+ * @param {HTMLAnchorElement} link
+ * @return {Object} object with the values to call ga send event command.
+ */
+function getData(link) {
+  const data = Object.assign(
+    defaultsForLink(link),
+    link.dataset,
+    compulsoryFields(link),
+  );
+
+  return EVENT_FIELDS.reduce((memo, key) => {
+    /* eslint no-param-reassign: ["error", { "props": false }] */
+    if (key in data) {
+      memo[key] = data[key];
+    }
+    return memo;
+  }, {});
+}
+
+/**
+ * Returns the arguments to call ga command when the link has been clicked.
+ *
+ * @param {HTMLAnchorElement} link element to get the data from.
+ * @return {Array<String, String, Object>} arguments for ga method.
+ */
+function commandArguments(link) {
+  return [SEND_COMMAND, EVENT_HIT_TYPE, getData(link)];
+}
+
+const gaInternal = [];
+const ga = window.ga || function (...args) {
+  gaInternal.push(args);
+  console.log(...args);
+
+  const last = args.pop();
+
+  if (typeof last.hitCallback === 'function') {
+    setTimeout(last.hitCallback.bind(last), 0);
+  }
+};
+
+/**
+ * Promisify ga
+ *
+ * @param {...any} args
+ * @return {Promise} a promise that resolves whenever the hitCallback is called.
+ */
+function gaPromise(...args) {
+  let obj = {};
+  let oldCallback = () => { };
+
+  if (typeof args[args.length - 1] === 'object') {
+    obj = args.pop();
+  }
+
+  if (typeof obj.hitCallback === 'function') {
+    oldCallback = obj.hitCallback;
+  }
+
+  return new Promise((resolve) => {
+    obj.hitCallback = (...cbArgs) => {
+      oldCallback(...cbArgs);
+      resolve(...cbArgs);
+    };
+    ga(...args, obj);
+  });
+}
+
+function isScrollableLink(link) {
+  const [url, id] = link.href.split('#');
+
+  return window.location.href.indexOf(url) > 0 && id;
+}
+
+/**
+ * Scrolls to element anchored by the given link.
+ *
+ * @param {HTMLAnchorElement} link
+ */
+function scrollTo(link) {
+  const [, id] = link.href.split('#');
```

Comment:
the `scrollTo` behavior is already implemented inside of `index.js::setupSmoothScroll()` ([here](https://github.com/src-d/landing/blob/master/src/js/index.js#L62))
We could maybe export a `scrollTo` implementation and use it in both places to avoid duplicating code. Would you agree on that?